torch/csrc/api/src/nn/modules/loss.cpp

```diff
@@ -49,5 +49,18 @@ Tensor CosineEmbeddingLossImpl::forward(
   return F::cosine_embedding_loss(input1, input2, target, options);
 }
 
+// ============================================================================
+
+SoftMarginLossImpl::SoftMarginLossImpl(
+    const torch::nn::SoftMarginLossOptions& options_) : options(options_) {}
+
+void SoftMarginLossImpl::pretty_print(std::ostream& stream) const {
+  stream << "torch::nn::SoftMarginLoss(reduction=" << options.reduction() << ")";
```

Comment:
Similarly here, we can add a test for `pretty_print`, and as part of the test we can test that `SoftMarginLoss(/*reduction=*/Reduction::Mean)` (i.e. the implicit constructor of `SoftMarginLossOptions`) works :D